demos/tests/cases.py

```diff
@@ -40,13 +51,15 @@ def models_lst_path(self, source_dir):
     def fixed_args(self, source_dir, build_dir):
         return [str(build_dir / self._name)]
 
-class PythonDemo:
-    def __init__(self, subdirectory, test_cases):
+class PythonDemo(Demo):
+    def __init__(self, subdirectory, device_keys, test_cases):
         self.subdirectory = 'python_demos/' + subdirectory
 
         self._name = subdirectory.replace('/', '_')
 
         self.test_cases = test_cases
+        
+        self.device_keys = device_keys
```

Comment:
These assignments should probably happen in the same order that the parameters are listed.